src/api/app/helpers/maintenance_helper.rb

```diff
@@ -140,7 +140,7 @@ def release_package_create_main_package(request, source_package, target_package_
     upload_params[:cmd] = 'commitfilelist'
     upload_params[:noservice] = '1'
     upload_params[:requestid] = request.number if request
-    upload_path = "/source/#{URI.escape(target_project.name)}/#{URI.escape(base_package_name)}"
+    upload_path = "/source/#{CGI.escape(target_project.name)}/#{CGI.escape(base_package_name)}"
```

Comment:
be careful with CGI.escape on path components.

```
URI.escape('devel:libraries:c_c++')
=> "devel:libraries:c++"

curl 'src-opensuse:5352/source/devel:libraries:c_c++' => 
<directory>
....

CGI.escape('devel:libraries:c_c++')
=> "devel%3Alibraries%3Ac%2B%2B"

curl 'src-opensuse:5352/build/devel%3Alibraries%3Ac%2B%2B'
<status code="404">
  <summary>project 'devel:libraries:c++' does not exist</summary>
  <details>404 project 'devel:libraries:c++' does not exist</details>

```

Reply:
This seems to be a copy'n'paste problem on your side:
```
CGI.escape('devel:libraries:c_c++')
=> "devel%3Alibraries%3Ac_c%2B%2B"
curl 'src-opensuse:5352/build/devel%3Alibraries%3Ac_c%2B%2B'
<directory>
...
```

Reply:
Right, it's c_c++ :)

Reply:
Still be careful :)

Reply:
The problem case is spaces in path components:
```
CGI.escape('foo bar')
=> "foo+bar"
```
That won't fly as path component.